### PR TITLE
Fix issue 56 with updates to require an alias

### PR DESCRIPTION
**Pull request by thenewguy**
Supersedes https://github.com/tox-dev/tox-docker/pull/58

### CODE REVIEW

File: tox_docker.py [hunk 1]
```diff
@@ -110,6 +110,8 @@ def getint(reader, key):
             })
         if reader.getstring("ports"):
             image_configs[image]["ports"] = reader.getlist("ports")
+        if reader.getstring("links"):
+            image_configs[image]["links"] = [link.strip() for link in reader.getstring("links").split(' ') if link.strip()]
```
Comment (dcrosta) [inline tox_docker.py:114]: can we use `reader.getlist()` (I think that's what it's called) -- that lets you specify it in the typical tox way

File: test_links.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            if nginx_container is None and "nginx" in container.attrs['Config']['Image']:
+                nginx_container = container
+            
+            if all([registry_container, nginx_container]):
+                break
+
+        self.assertIsNotNone(registry_container)
+        self.assertIsNotNone(nginx_container)
+        
+        self.assertEqual(registry_container.exec_run("ping -c 1 apache")[0], 0)
```
Comment (dcrosta) [inline test_links.py:71]: does `curl` not work here (maybe it's not in the image, I guess?)

File: tox_docker.py [hunk 3]
```diff
@@ -110,6 +110,8 @@ def getint(reader, key):
             })
         if reader.getstring("ports"):
             image_configs[image]["ports"] = reader.getlist("ports")
+        if reader.getstring("links"):
+            image_configs[image]["links"] = reader.getlist("links")
```
Comment (dcrosta) [inline tox_docker.py:114]: It's probably better to validate the format (and values, if possible?) of the links here, in terms of which phase of the tox run gets the failure.

File: test_links.py (hunk 2 above)
Comment (thenewguy) [reply test_links.py:71]: It wasn't in the image.  Had trouble picking images from the official repo with networking tools

File: tox_docker.py (hunk 3 above)
Comment (thenewguy) [reply tox_docker.py:114]: Agreed - will see what I can work out

Comment (thenewguy) [reply tox_docker.py:114]: validation requires access to the docker containers - do you care about validating twice? could partially validate upfront and then let it run again without too much trouble

**Comment by thenewguy** [comment]
@dcrosta does tox have a registry we can pull against?

I need to test an image from a registry with a port specified but dockerhub uses a random high port according to irc and gives a 503 on port 443 when used directly

I.e. registry.hub.docker.com/library/httpd:2.4-alpine but I can't check for registry.hub.docker.com:443/library/httpd:2.4-alpine

**Comment by thenewguy** [comment]
Right now the port specified in the url would break the partition call - I wanted to add a test to ensure conformance if possible

File: tox_docker.py (hunk 3 above)
Comment (dcrosta) [reply tox_docker.py:114]: Yeah, if we don't know the names of all the other images, we can skip that part, but at least the format validation can happen here.

**Comment by dcrosta** [comment]
I think this is a step in a good direction, even if there are some rough edges around registry URLs on custom ports. I think it can ship as-is.

I'm about to put up a proposal for new config syntax in 2.0 that can handle this more gracefully.

**Comment by dcrosta** [comment]
(See #65)

**Comment by thenewguy** [comment]
Sounds great!  This one is RFC

**Comment by dcrosta** [comment]
One last thing: could you squash all these commits together? I'll merge it after that.

**Comment by thenewguy** [comment]
@dcrosta I will but I am covered up for awhile.  If you have time would you mind doing it so the merge isn't delayed?

**Comment by dcrosta** [comment]
merged and squashed in 916f42552dae29d08f7e9ae29b5ca09f492b927a -- thanks!

**Pull request closed without merging**